contracts/StakeManager.sol

```diff
@@ -222,9 +222,18 @@ contract StakeManager is Ownable {
         if (deltaTime < MIN_LOCKUP_PERIOD || deltaTime > MAX_LOCKUP_PERIOD) {
             revert StakeManager__InvalidLockTime();
         }
-        _mintInitialMP(account, _timeToIncrease, 0);
-        //update account storage
+
+        uint256 mpToMint = _getMaxMPToMint(
+            _getMPToMint(account.balance, _timeToIncrease),
+            account.balance,
+            account.initialMP,
+            account.currentMP
+        );
+
+        totalSupplyMP += mpToMint;
+        account.currentMP += mpToMint;
         account.lockUntil = lockUntil;
+        account.lastMint = block.timestamp;
```

Comment:
The are storage updates that were otherwise happening in `_mintInitialMP()`.

Question @3esmit: I think we need to update `currentEpoch.totalSupply` as well no?
If yes, then we could potentially just leverage `_mintMP()` instead.